src/ui/units/connections/components/custom-forms/Yadocs/components/DocsList.tsx

```diff
@@ -1,31 +1,56 @@
 import React from 'react';
 
+import {List} from '@gravity-ui/uikit';
 import block from 'bem-cn-lite';
-import {YadocsAddSectionState} from 'ui/units/connections/store';
+import type {YadocItem, YadocsAddSectionState} from 'ui/units/connections/store';
 
-import {AddSection} from '../components';
-import type {AddYandexDoc, UpdateAddSectionState} from '../types';
+import {AddSection, YadocListItemView} from '../components';
+import type {AddYandexDoc, HandleItemClick, UpdateAddSectionState, YadocListItem} from '../types';
 
 const b = block('conn-form-yadocs');
+const ITEM_HEIGHT = 52;
 
 type Props = {
     addSectionState: YadocsAddSectionState;
+    items: YadocItem[];
+    selectedItemIndex: number;
     addYandexDoc: AddYandexDoc;
+    clickListItem: HandleItemClick;
     updateAddSectionState: UpdateAddSectionState;
 };
 
 export const DocsList = (props: Props) => {
-    const {addSectionState, addYandexDoc, updateAddSectionState} = props;
+    const {
+        addSectionState,
+        items,
+        selectedItemIndex,
+        addYandexDoc,
+        clickListItem,
+        updateAddSectionState,
+    } = props;
+
+    const renderItem = React.useCallback((item: YadocListItem) => {
```

Comment:
Is there a reason why this function should be inside the component and not outside?

Reply:
I suppose it's pretty clear to keep items render method near List component.
There are store actions outside this component. This method handle UI